feat(PermissionForms): add classes and students tables, add related APIs and policies

pjanik opened the pull request:
https://www.pivotaltracker.com/story/show/187612778
https://www.pivotaltracker.com/story/show/187595842

This PR adds tables for classes and students, + all the related APIs and policies.

I have added a new endpoint to the Teacher API, as a simple list of classes seemed straightforward and generally useful. However, an API to obtain a list of class students along with their permission forms ended up in the Permission Forms API instead of the Class API. In fact, the Class API already has an endpoint that renders most of the necessary information, but it provides more than we need and does not actually include the list of permission forms. So, I thought it would not make sense to return permission forms in every context and that it would be better to have more page-oriented APIs that handle tasks efficiently. I believe this will also be easier to maintain in the long term.

I also updated policies related to permission forms to allow only admin and project admin, but this might change in the future: https://concord-consortium.slack.com/archives/C0M5CM1RA/p1718911188565409

dougmartin commented on rails/react-components/src/library/components/permission-forms-v2/students-tab/students-tab.tsx:
```diff
@@ -69,33 +69,36 @@ export default function StudentsTab() {
+              teachers.map(teacher => {
+                const active = selectedTeacherId === teacher.id;
+                return (
+                  <>
```
I'm guessing this is generating a key warning.  You can use `<React.Fragment key={teacher.id}>` here to fix it.

pjanik replied:
Thanks, fixed in recent commits.

dougmartin commented on rails/react-components/src/library/components/permission-forms-v2/students-tab/classes-table.tsx:
```diff
@@ -8,14 +12,59 @@ interface IProps {
+            const active = selectedClassId === classInfo.id;
+            return (
+              <>
+                <tr key={classInfo.id} className={clsx({ [css.activeRow]: active })}>
```
I think this key needs to be hoisted into the parent fragment.

pjanik replied:
Thanks, fixed in recent commits.

dougmartin reviewed (approved):
Looks good :+1:   You have some failing tests but the code changes are good minus some adjustments of the map keys.

pjanik commented:
I've addressed the key issues, and fixed spec. Also, I restored access for project researchers, as it seems this will be necessary (more info here: https://concord-consortium.slack.com/archives/C0M5CM1RA/p1718981221934589). So, merging this PR.
